packages/transform/src/template-to-typescript.ts

```diff
@@ -77,16 +79,39 @@ export function templateToTypescript(
       }
     }
 
-    function emitTemplateBoilerplate(emitBody: () => void): void {
-      emit.text(`({} as typeof import("${typesPath}")).template(function`);
-      emit.synthetic(typeParams);
-      emit.text('(𝚪');
+    function emitTemplateBoilerplate(emitBody: () => void, useJsDoc: boolean): void {
+      if (useJsDoc) {
+        emit.text(
+          `${
+            contextType ? '/** @type {unknown} */ (' : ''
+          }/** @type {import("${typesPath}")} */ ({}).template(function(`
+        );
+      } else {
+        emit.text(`({} as typeof import("${typesPath}")).template(function`);
+        emit.synthetic(typeParams); // investigate JSDoc support for this as a followup
```

Comment:
I'm inclined to say it's probably ok to just not support type params for JS sources. The alternative is getting into parsing the JSDoc for the class ourselves and sorting through `@template` pragmas and their friends, which is possible but definitely a nontrivial amount of work.

Reply:
Sounds good, thanks for the feedback! I'll remove the investigation comment then